internal/pkg/skuba/addons/cilium.go

```diff
@@ -87,235 +79,524 @@ func (ciliumCallbacks) afterApply(addonConfiguration AddonConfiguration, skubaCo
+    k8s-app: cilium
+    kubernetes.io/cluster-service: "true"
   name: cilium
   namespace: kube-system
 spec:
   revisionHistoryLimit: 3
-  updateStrategy:
-    type: "RollingUpdate"
-    rollingUpdate:
-      # Specifies the maximum number of Pods that can be unavailable during the update process.
-      # The current default value is 1 or 100% for daemonsets; Adding an explicit value here
-      # to avoid confusion, as the default value is specific to the type (daemonset/deployment).
-      maxUnavailable: "100%"
   selector:
     matchLabels:
       k8s-app: cilium
       kubernetes.io/cluster-service: "true"
   template:
     metadata:
-      labels:
-        k8s-app: cilium
-        kubernetes.io/cluster-service: "true"
       annotations:
-        {{.AnnotatedVersion}}
         # This annotation plus the CriticalAddonsOnly toleration makes
         # cilium to be a critical pod in the cluster, which ensures cilium
         # gets priority scheduling.
         # https://kubernetes.io/docs/tasks/administer-cluster/guaranteed-scheduling-critical-addon-pods/
-        scheduler.alpha.kubernetes.io/critical-pod: ''
-        scheduler.alpha.kubernetes.io/tolerations: >-
-          [{"key":"dedicated","operator":"Equal","value":"master","effect":"NoSchedule"}]
+        scheduler.alpha.kubernetes.io/critical-pod: ""
+        scheduler.alpha.kubernetes.io/tolerations: '[{"key":"dedicated","operator":"Equal","value":"master","effect":"NoSchedule"}]'
+      labels:
+        k8s-app: cilium
+        kubernetes.io/cluster-service: "true"
     spec:
-      serviceAccountName: cilium
-      initContainers:
-      - name: install-cni-conf
-        image: {{.CiliumImage}}
-        command:
-          - /bin/sh
-          - "-c"
-          - "cp -f /etc/cni/net.d/10-cilium-cni.conf /host/etc/cni/net.d/10-cilium-cni.conf"
-        volumeMounts:
-        - name: host-cni-conf
-          mountPath: /host/etc/cni/net.d
-      - name: install-cni-bin
-        image: {{.CiliumImage}}
-        command:
-          - /bin/sh
-          - "-c"
-          - "cp -f /usr/lib/cni/* /host/opt/cni/bin/"
-        volumeMounts:
-        - name: host-cni-bin
-          mountPath: /host/opt/cni/bin/
-      - name: clean-cilium-state
-        image: {{.CiliumInitImage}}
-        imagePullPolicy: IfNotPresent
+      containers:
+      - args:
+        - --config-dir=/tmp/cilium/config-map
         command:
-        - cilium-init
+        - cilium-agent
         env:
-        - name: CLEAN_CILIUM_STATE
+        - name: K8S_NODE_NAME
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: spec.nodeName
+        - name: CILIUM_K8S_NAMESPACE
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.namespace
+        - name: CILIUM_FLANNEL_MASTER_DEVICE
           valueFrom:
             configMapKeyRef:
-              key: clean-cilium-state
+              key: flannel-master-device
```

Comment:
I am confused, why are there flannel keys around?

Reply:
Because Cilium has a feature of bridging on top of Flannel, it's just copied from upstream. If you feel confused by that, I can remove it.